erroranalysis/tests/test_surrogate_error_tree.py

```diff
@@ -41,6 +47,44 @@ def test_surrogate_error_tree_int_categorical(self):
         run_error_analyzer(model, X_test, y_test, list(X_train.columns),
                            categorical_features)
 
+    def test_large_data_surrogate_error_tree(self):
+        # validate tree trains quickly for large data
+        X_train, y_train, X_test, y_test, _ = \
+            create_binary_classification_dataset(100)
+        feature_names = list(X_train.columns)
+        model = create_sklearn_random_forest_regressor(X_train, y_train)
+        for _ in range(16):
+            X_test = pd.concat([X_test, X_test], ignore_index=True)
+            y_test = np.concatenate([y_test, y_test])
+        assert X_test.shape[0] > 1000000
+        t0 = time.time()
+        categorical_features = []
+        model_analyzer = ModelAnalyzer(model, X_test, y_test,
+                                       feature_names,
+                                       categorical_features)
+        max_depth = 3
+        num_leaves = 31
+        min_child_samples = 20
+        cat_ind_reindexed = []
+        diff = model_analyzer.get_diff()
+        surrogate = create_surrogate_model(model_analyzer,
+                                           X_test,
+                                           diff,
+                                           max_depth,
+                                           num_leaves,
+                                           min_child_samples,
+                                           cat_ind_reindexed)
+        t1 = time.time()
+        execution_time = t1 - t0
+        print(execution_time)
+        # assert we don't take too long to train the tree on 1 million rows
+        # note we train on >1 million rows in ~1 second
+        assert execution_time < 5
+        model_json = surrogate._Booster.dump_model()
```

Comment:
In my experience assert on execution time can be really flaky. The execution time depends on a lot of factors like avaible CPU, memory and may not be consistently less than 5 seconds here. Maybe the above print is suffice.

Reply:
I expanded it to 20 seconds then - certainly, if it only takes a second to train on >1 million rows, and suddenly it takes 20, then something must be wrong.  We can always increase the time more later if we see it failing.